python_ta/checkers/redundant_or_impossible_condition_checker.py

```diff
@@ -0,0 +1,117 @@
+"""
+Check for redundant If or While conditions in functions based on z3 constraints
+"""
+
+from typing import Any, Union
+
+try:
+    import z3
+
+    z3_dependency_available = True
+except ImportError:
+    z3 = Any
+    z3_dependency_available = False
+
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+
+class RedundantOrImpossibleConditionChecker(BaseChecker):
+    name = "redundant-condition"
+    msgs = {
+        "R9900": (
+            """This conditional statement is already true given the preceding code;
+             consider removing redundant check.""",
+            "redundant-condition",
+            "Used when an If or While statement is tautologically true based on preceding Z3 constraints."
+            " Enabled when Z3 option is on.",
```

Comment:
Enabling and disabling is a configuration option, not automatic. But a good thing to note here is "Requires the z3 configuration option to be True."